### PR TITLE
chore: Add sonarcloud configurations

### PR DESCRIPTION
**Pull request by onuzbee**
# Closes #
Closes DI-1381. 

Adds sonarcloud configurations. Related PR in `twilio-cli-core` with tested main push: [Link](https://github.com/twilio/twilio-cli-core/pull/136)

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,8 @@
+sonar.projectKey=twilio_twilio-cli
```
Comment (ravali-rimmalapudi) [inline sonar-project.properties:1]: [Q] Where are we using these properties?

Comment (onuzbee) [reply sonar-project.properties:1]: This key is generated by Sonar dashboard, which needs to be a 1:1 mapping between repositories. 
Helps sonar to link a CI build to it's dashboard view.

File: sonar-project.properties [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+sonar.projectName=twilio-cli
+sonar.organization=twilio
+
+sonar.sources=src/
+sonar.tests=test/
+# sonar.test.exclusions= < No exclusions currently >
```
Comment (ravali-rimmalapudi) [inline sonar-project.properties:7]: nit: If not required, we can remove this.

File: .travis.yml [hunk 3]
```diff
@@ -31,3 +34,7 @@ notifications:
...
+addons:
+  sonarcloud:
+    token:
+      secure: HoD6oJ03Y60apDszjFZ8AN+ujnkbQAPM41bBb/YNNMtgslfnJPp63bQy5m0hoCsTVRoPiZvuZRpfxQpcJLmlEScY4ulQI7DRf+FFh6/2B80+kl1FRLsyar0/JtZvn/LDsP5GjksIZUD2cdPho2rRnmGiS9bmrKIc4YRKKxMO/zi33EUPgiwea23rCB/6/CK6AnAyD7ufwM/ldewqKiU5T67IbAQM2mKEMqyM+oPSx/HL6f9NrtpenZs9pP3y8oRCCty8yP5BOyoLqOJvn0lRqdAAL5S8Jsrb6NB7Mk+lrU4AcfcgEmDGbj+wKFdV/xlqZcCDGDvytqGocGsQZLwbaNlOzSTkoJVvzy1crMxySsqVFoGusISPXA3iJgBQn08E8bwSI1SWDfdvQTFZFYcoc/1OWNutH6EvoHC9uM9rmIRJPVDqGeBevlJq7vffspnuwpWZeH6hWQt7kHBGxQSYtCyOUkFU0IntRFiX2KvyP38rRIlfeoDM2fqaD/NUg0+C77YYhi9on74mQrw8EEjq5QErht458shfjiPPoq8PHpN2WNXjJ8lRHc6h8eqNzX0Uz2ooPOvxgFnyjergBbe5KADQkPLhJehZQhNtHiAfAX5dtkP8IJizu0tpwrTkMWrMMT51nbqL/zOtH6yflr1+AjKrLM7YFRSj8by3yokNzI4=
```
Comment (ravali-rimmalapudi) [inline .travis.yml:40]: [Q] Is this token generated from sonar?

**Review by ravali-rimmalapudi — COMMENTED** [review]
Looks Good.

**Review by ravali-rimmalapudi — APPROVED** [review]

File: .travis.yml (hunk 3 above)
Comment (onuzbee) [reply .travis.yml:40]: Used travis encryption for sonar cloud token 

```
travis encrypt --add addons.sonarcloud.token <SONAR_TOKEN>
```

File: sonar-project.properties (hunk 2 above)
Comment (onuzbee) [reply sonar-project.properties:7]: I've deliberately added this to avoid checking documentation back again for skipping some specific test branches.

**Pull request merged**